src/options.rs

```diff
@@ -409,15 +390,24 @@ impl Options {
             && atty::is(atty::Stream::Stdin);
 
         let mut builder = conn_params(&tmp.conn);
-        if let (Some(Command::Authenticate(auth)), None) = (&tmp.subcommand, &tmp.query) {
-            if builder.is_err() {
-                builder = Ok(Builder::new());
-                load_tls_options(&tmp.conn, builder.as_mut().unwrap())?;
+
+        // Special case for `edgedb instance link`
+        if let (Some(Command::Instance(i)), None) = (
```

Comment:
Oh, I've overlooked this previosly, but as we here: we don't need to check tuple here. `tmp.subcommand.is_some() && tmp.query.is_some()` is prohibited and it's checked below in `let subcommand =`.

Reply:
Fixed, so that we check and fail early before doing anything else.